### PR TITLE
[FIX] react hooks

### PR DESCRIPTION
**Pull request by farazatarodi**
### Description

Adds the eslint plugin for React hook rules.

#### Screenshot before this PR

NA

#### Screenshot after this PR

NA

### Breaking changes

NA

### CODE REVIEW

**Review by jelledc — DISMISSED** [review]

**Review by KristofColpaert — DISMISSED** [review]

**Comment by lorgan3** [comment]
How well is this tested? Changing the dependency arrays is very likely to change the behaviour of those components

**Comment by farazatarodi** [comment]
@lorgan3 tested all of them. But an OTS MC is always welcome. I might have missed something.

**Review by JorenSaeyTL — APPROVED** [review]

**Review by lowiebenoot — APPROVED** [review]
I think this is quite risky to do though.

**Comment by farazatarodi** [comment]
Reverted the commit that "fixes" the dependencies as it is impossible to guarantee the prevention of bugs.

**Comment by JorenSaeyTL** [comment]
Good job but I think only the dependencies in the `useEffect`s are risky. I guess better safe then sorry 😉

**Review by JorenSaeyTL — APPROVED** [review]

File: src/components/datagrid/DataGrid.tsx [hunk 1]
```diff
@@ -74,7 +74,7 @@ export const DataGrid: DatagridComponent = ({
             .reduce((accumulatedChildWidth, currentChildWidth) => accumulatedChildWidth + currentChildWidth, 0),
         )
         .reduce((maxRowWidth, currentRowWidth) => (currentRowWidth > maxRowWidth ? currentRowWidth : maxRowWidth), 0),
-    [rowNodes, rowNodes.size],
+    [rowNodes],
```
Comment (lowiebenoot) [inline src/components/datagrid/DataGrid.tsx:77]: I think this useMemo will now run only once, as `rowNodes` never seems to be changed and is always the same Map. That's whay the `rowNodes.size` was added I guess.

File: src/components/tab/TabGroup.tsx [hunk 2]
```diff
@@ -54,7 +54,9 @@ const TabGroup: GenericComponent<TabGroupProps> = ({ children, className, size,
   useEffect(() => {
     smoothScroll.polyfill();
     scrollContainerRef.current?.addEventListener('scroll', handleScroll);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (lorgan3) [inline src/components/tab/TabGroup.tsx:57]: This one can is probably not required

File: src/components/tooltip/Tooltip.tsx [hunk 3]
```diff
@@ -87,6 +87,7 @@ const TooltippedComponent: GenericComponent<TooltippedComponentProps> = ({
   ComposedComponent,
   ...other
 }) => {
+  // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (lorgan3) [inline src/components/tooltip/Tooltip.tsx:90]: Same for this one

File: src/components/wysiwygEditor/WysiwygEditor.tsx [hunk 4]
```diff
@@ -127,8 +127,10 @@ const WysiwygEditor: GenericComponent<WysiwygEditorProps> = ({
       if (!onKeyDown || !editorRef?.current?.wrapper) {
         return;
       }
+      // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (lorgan3) [inline src/components/wysiwygEditor/WysiwygEditor.tsx:130]: And this one

File: src/components/tooltip/Tooltip.tsx (hunk 3 above)
Comment (farazatarodi) [reply src/components/tooltip/Tooltip.tsx:90]: @lorgan3 Why?

**Pull request closed without merging**